src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/CacheHealthDecider.java

```diff
@@ -130,7 +134,14 @@ private void configureModifyCacheActionPriority() {
    * signals going forward.
    */
   private Action computeBestAction(final NodeKey esNode, final ResourceEnum cacheType) {
-    return getAction(ModifyCacheMaxSizeAction.NAME, esNode, cacheType, true);
+    Action action = null;
+    if (canUseMoreHeap(esNode)) {
+      action = getAction(ModifyCacheMaxSizeAction.NAME, esNode, cacheType, true);
+    } else {
+      PerformanceAnalyzerApp.RCA_RUNTIME_METRICS_AGGREGATOR.updateStat(
```

Comment:
I am wondering if we should also add a log statement here in addition to the metric. Thoughts?

Reply:
We don't want to emit the log continuously and in a cluster with jvm pressure, we will emit it quite a few times.